jsonrpc/server_test.go

```diff
@@ -439,3 +440,83 @@ func TestRequestValidation(t *testing.T) {
 		})
 	}
 }
+
+func TestMaxRequestPerIPPerSec(t *testing.T) {
+	// this is the number of requests the test will execute
+	// it's important to keep this number with an amount of
+	// requests that the machine running this test is able
+	// to execute in a single second
+	const numberOfRequests = 100
+	// the number of workers are the amount of go routines
+	// the machine is able to run at the same time without
+	// consuming all the resources and making the go routines
+	// to affect each other performance, this number may vary
+	// depending on the machine spec running the test.
+	// a good number to this generally is a number close to
+	// the number of cores or threads provided by the CPU.
+	const workers = 12
+	// it's important to keep this limit smaller than the
+	// number of requests the test is going to perform, so
+	// the test can have some requests rejected.
+	const maxRequestsPerIPAndSecond = 20
+
+	cfg := getSequencerDefaultConfig()
+	cfg.MaxRequestsPerIPAndSecond = maxRequestsPerIPAndSecond
+	s, m, _ := newMockedServerWithCustomConfig(t, cfg)
+	defer s.Stop()
+
+	// since the limitation is made by second,
+	// the test waits 1 sec before starting because request are made during the
+	// server creation to check its availability. Waiting this second means
+	// we have a fresh second without any other request made.
+	time.Sleep(time.Second)
+
+	// create a wait group to wait for all the requests to return
+	wg := sync.WaitGroup{}
+	wg.Add(numberOfRequests)
+
+	// prepare mocks with specific amount of times it can be called
+	// this makes us sure the code is calling these methods only for
+	// allowed requests
+	times := int(cfg.MaxRequestsPerIPAndSecond)
+	m.DbTx.On("Commit", context.Background()).Return(nil).Times(times)
+	m.State.On("BeginStateTransaction", context.Background()).Return(m.DbTx, nil).Times(times)
+	m.State.On("GetLastL2BlockNumber", context.Background(), m.DbTx).Return(uint64(1), nil).Times(times)
+
+	// prepare the workers to process the requests as long as a job is available
+	jobs := make(chan int)
+	// put each worker to work
+	for i := 0; i < workers; i++ {
+		// each worker works in a go routine to be able to have many
+		// workers working concurrently
+		go func() {
+			// a worker keeps working indefinitely looking for new jobs
+			for {
+				// waits until a job is available
+				<-jobs
+				// send the request
+				_, err := s.JSONRPCCall("eth_blockNumber")
+				// if the request works well or gets rejected due to max requests per sec, it's ok
+				// otherwise we stop the test and log the error.
+				if err != nil && err.Error() != "429 - You have reached maximum request limit." {
+					require.NoError(t, err)
+				}
+				// registers in the wait group a request was executed and has returned
+				wg.Done()
+			}
+		}()
+	}
+
+	// add jobs to notify workers accordingly to the number
+	// of requests the test wants to send to the server
+	for i := 0; i < numberOfRequests; i++ {
+		jobs <- i
```

Comment:
`Job` is not a buffered channel, so these requests will not be concurrent. Is that the desired behaviour?